src/toil/wdl/wdltoil.py

```diff
@@ -2210,6 +2248,54 @@ def can_mount_proc(self) -> bool:
         """
         return "KUBERNETES_SERVICE_HOST" not in os.environ
 
+    def ensure_mount_point(self, file_store: AbstractFileStore, mount_spec: Dict[str, int]) -> Dict[str, str]:
+        """
+        Ensure the mount point sources are available.
+
+        Will check if the mount point source has the requested amount of space available.
+
+        Note: We are depending on Toil's job scheduling backend to error when the sum of multiple mount points disk requests is greater than the total available
+        For example, if a task has two mount points request 100 GB each but there is only 100 GB available, the df check may pass
+        but Toil should fail to schedule the jobs internally
+
+        :param mount_spec: Mount specification from the disks attribute in the WDL task. Is a dict where key is the mount point target and value is the size
+        :param file_store: File store to create a tmp directory for the mount point source
+        :return: Dict mapping mount point target to mount point source
+        """
+        logger.debug("Detected mount specifications, creating mount points.")
+        mount_src_mapping = {}
+        # Create one tmpdir to encapsulate all mount point sources, each mount point will be associated with a subdirectory
+        tmpdir = file_store.getLocalTempDir()
+
+        # The POSIX standard doesn't specify how to escape spaces in mount points and file system names
+        # The only defect of this regex is if the target mount point is the same format as the df output
+        # It is likely reliable enough to trust the user has not created a mount with a df output-like name
+        regex_df = re.compile(r".+ \d+ +\d+ +(\d+) +\d+% +.+")
+        try:
+            for mount_target, mount_size in mount_spec.items():
```

Comment:
Rather than doing this as a loop over each requested mount point (except the `local-disk`/no-mount-point ones that set the size of the working directory, whicvh never made it into `mount_spec`), and checking for space for each individually, we should sum up all the space needed and check for at least that much total space. We know all the mount points and also the task working directory are going to come from the same underlying filesystem where the Toil work directory is. So we only need to ask about the free space on that filesystem once.

Then we would no longer have the two-100GB-mount-points problem; we would know that we need 200GB total and only have say 150.

We can probably just read the job's disk requirement (I think `self.requirements.disk`) and make sure `df` shows that much space. Then we don't need two copies of the logic to sum up the total of all the mount points.